### ISSUE TITLE
False negative for enum constants in UndocumentedPublic* rule group

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#7218]
## Steps to Reproduce
```kotlin
public enum class PublicEnum {
	Foo,
	Bar,
}
```

Configuration:
```gradle.kts
detekt.allRules = true
// Does not contain relevant lines, just a few rules deactivated or fine tuned.
detekt.config.from(rootProject.file("config/detekt/detekt.yml"))
dependencies { detektPlugins(libs.detekt.rules.libraries) }
```

## Observed Behavior
```console
src\main\kotlin\com\example\PublicEnum.kt:21:15: PublicEnum is missing required documentation. [UndocumentedPublicClass]
```

## Expected Behavior
Findings for each of the undocumented enum constants. They're public API as well, accessible as `PublicEnum.Foo`.
If
```kotlin
public companion object {
    public val Foo: PublicClass = ...
}
```
needs to be documented (`The property Foo is missing documentation. [UndocumentedPublicProperty]`), so does these enum constants `Foo` and `Bar` as they're essentially "just syntactic a shorthand" for the same.

## Context
I'm relying on detekt to ensure that all my public API surface is documented.

This repro might be a trivial case, but imagine an enum with 10-12 values, and no documentation.

 * Related: https://github.com/detekt/detekt/issues/7217

## Your Environment
* Version of detekt used: 1.23.5
* Version of Gradle used (if applicable): 8.7
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows 10 / GitHub Actions Ubuntu 22.04
* Link to your project (if it's a public repository): N/A

**Comment by TWiStErRob** [comment]
Documentation alone is cumbersome :) if someone enables this rule, they probably want it?

**Comment by BraisGabin** [comment]
Can you open a new issue describing your use case and your proposal to fix it?
